dds/DCPS/XTypes/Utils.cpp

```diff
@@ -1206,6 +1206,376 @@ DDS::ReturnCode_t set_enum_value(
+        }
+      }
+      break;
+    case TK_UINT16:
+      {
+        CORBA::UInt16 v;
+        rc = src->get_uint16_value(v, id);
+        if (rc != DDS::RETCODE_OK) {
+          value = v;
+        }
+      }
+      break;
+    case TK_UINT32:
+      {
+        CORBA::UInt32 v;
+        rc = src->get_uint32_value(v, id);
+        if (rc != DDS::RETCODE_OK) {
+          value = v;
+        }
+      }
+      break;
+    case TK_UINT64:
+      rc = src->get_uint64_value(value, id);
+      break;
+    }
+    return rc;
+  }
+
+  DDS::ReturnCode_t set_uint_like(
+    DDS::DynamicData_ptr dest, DDS::MemberId id, DDS::TypeKind kind, CORBA::UInt64 value)
+  {
+    switch (kind) {
+    case TK_UINT8:
+      return dest->set_uint8_value(id, value);
+    case TK_UINT16:
+      return dest->set_uint16_value(id, value);
+    case TK_UINT32:
+      return dest->set_uint32_value(id, value);
+    case TK_UINT64:
+      return dest->set_uint64_value(id, value);
+    default:
+      return DDS::RETCODE_BAD_PARAMETER;
+    }
+  }
+}
+
+DDS::ReturnCode_t copy(DDS::DynamicData_ptr dest, DDS::DynamicData_ptr src)
+{
+  if (dest == src) {
+    return DDS::RETCODE_OK;
+  }
+
+  DDS::DynamicType_var src_type = src->type();
+  DDS::DynamicTypeMembersById_var src_members_var;
+  DDS::ReturnCode_t rc = src_type->get_all_members(src_members_var);
+  if (rc != DDS::RETCODE_OK) {
+    return rc;
+  }
+  DynamicTypeMembersByIdImpl* src_members =
+    dynamic_cast<DynamicTypeMembersByIdImpl*>(src_members_var.in());
+
+  DDS::DynamicType_var dest_type = dest->type();
+  DDS::DynamicTypeMembersById_var dest_members_var;
+  rc = dest_type->get_all_members(dest_members_var);
+  if (rc != DDS::RETCODE_OK) {
+    return rc;
+  }
+  DynamicTypeMembersByIdImpl* dest_members =
+    dynamic_cast<DynamicTypeMembersByIdImpl*>(dest_members_var.in());
+
+  for (DynamicTypeMembersByIdImpl::const_iterator src_it = src_members->begin();
+      src_it != src_members->end(); ++src_it) {
+    const DDS::MemberId id = src_it->first;
+    const DynamicTypeMembersByIdImpl::const_iterator dest_it = dest_members->find(id);
+    if (dest_it == dest_members->end()) {
+      continue;
+    }
+
+    DDS::MemberDescriptor_var src_md;
+    rc = src_it->second->get_descriptor(src_md);
+    if (rc != DDS::RETCODE_OK) {
+      return rc;
+    }
+    DDS::DynamicType_var src_mem_type = get_base_type(src_md->type());
+    const DDS::TypeKind src_tk = src_mem_type->get_kind();
+
+    DDS::MemberDescriptor_var dest_md;
+    rc = dest_it->second->get_descriptor(dest_md);
+    if (rc != DDS::RETCODE_OK) {
+      return rc;
+    }
+    DDS::DynamicType_var dest_mem_type = get_base_type(dest_md->type());
+    const DDS::TypeKind dest_tk = dest_mem_type->get_kind();
+
+    if (src_tk != dest_tk) {
+      continue;
+    }
+
+    DDS::ReturnCode_t get_rc = DDS::RETCODE_OK;
+    DDS::ReturnCode_t set_rc = DDS::RETCODE_OK;
+
+    switch (src_tk) {
+    case TK_BOOLEAN:
+      {
+        CORBA::Boolean value;
```

Comment:
Yes I can. I have a few things to update in this code as it hasn't changed since I stopped working on it a few months ago, I just haven't gotten around to it yet.

Reply:
Yeah, I realize it's in draft - I was just recording some notes when I looked it at yesterday